src/encoding.ts

```diff
@@ -75,6 +75,11 @@ export interface Encoding {
    * stack order for stacked marks or order of data points in line marks.
    */
   order?: OrderFieldDef | OrderFieldDef[];
+
+  /**
+   * Tooltip adds basic string tooltip support.
+   */
+  tooltip?: FieldDef | ValueDef<string>;
```

Comment:
This could be `TextFieldDef | ConditionalValueDef<string|number|boolean>` just like text.  

(Since tooltip will be pretty similar to `text`, you might wanna move it to above `order`, right after `text`.)